src/app/InteractionModelEngine.h

```diff
@@ -545,6 +551,8 @@ class InteractionModelEngine : public Messaging::UnsolicitedMessageHandler,
 
     FabricTable * mpFabricTable;
 
+    CASESessionManager * mpCASESessionMgr = nullptr;
```

Comment:
Discussed offline.  The solution that the PR allows is to extend `ReadClient` and override `OnResubscriptionNeeded` to create a call-path that doesn't depend on `CASESessionManager` so that a null `CASESessionManager` can be passed at init.

That will work.  I have just requested that the instances of VerifyOrDie(GetCaseSessionManager != nullptr) be replaced with code that propagates the error if possible, as the point here is that `CASESessionManager` isn't _strictly_ required if one is willing to override this feature.